README.md

```diff
@@ -0,0 +1,84 @@
+
+## Description
+
+The main purpose of the `spark8t` toolkit is to provide a seemless, user-friendly interface
+to Spark functionalities over Kubernetes. As much for administator tasks (such as account registration)
+or data scientist functions (such as job submission or Spark interactive shell access). Various
+wrapper scripts allow for persistent (and user-friendly) configuration and execution of related tools.
+
+## Dependencies and Requirements
+
+ - *Kubernetes*
+ - *Apache Spark*
+
+## Installation
+
+Below we describe the essential steps on how to set up a Spark cluster together with the `spark8t` tool.
+
+(However note that most of the "hassle" desribed below can be saved, in case you choose to use the 
+[canonical/spark-client-snap](canonical/spark-client-snap) Snap installation, that would both install
+dependencies, both prepare critical parts of the environment for you.)
+
+### Kubernetes
+
+In order to be able to run Spark on Kubernetes, you'll sure need to have a Kubernetes cluster installed :-)
+
+A simple installation of a lightweight Kubernetes implementation (Canonical's `microk8s`) can
+be found in our [Discourse Spark
+Tutorial](https://discourse.charmhub.io/t/spark-client-snap-tutorial-setup-environment/8951)
+
+Keep in mind to set the following environment variable:
+
+ - `KUBECONFIG`: the location of the Kubernetes cluster configuration (typically: /home/spark/.kube/config)
+
+### Spark
+
+You will need to install Spark as instructed at the official [Apache Spark pages](https://spark.apache.org/downloads.html).
+
+Note that the `SPARK_HOME` environment variable must be correctly set to point to the location of your Spark installation
+
+### spark8t
+
+You could install the contents of this repository either by direct checkout, or using `pip` such as
+
+```
+pip insatll git+https://github.com/canonical/spark-k8s-toolkit-py.git
+```
+
+You'll need to add a mandatory configuration for the tool, which points to the OCI image to be used for the Spark workers.
+The configuration file must be called `spark-defaults.conf`, and could have a list of contents according to possible
+Spark-accepted command-line parameters. However the following specific one has to be defined:
+
+```
+spark.kubernetes.container.image=ghcr.io/canonical/charmed-spark:<version>
+```
+
+(See the [Spark ROCK releases GitHub page](https://github.com/canonical/charmed-spark-rock/pkgs/container/charmed-spark) for available versions)
+
+Then you would need to assign the correct values for the following `spark8t` environment variables:
+
+ - `SPARK_CONFS`: location of the `spark8t` configuration file
+ - `HOME`: the home of the Spark user (typically: `/home/spark`)
+ - `SPARK_USER_DATA`: the location of Spark user data, such as interactive shell history (typically: same as `HOME`)
+
+## Basic Usage
+
+`spark8t` is "built around" Spark itself, thus the usage is very similar to the known Spark client tools.
```

Comment:
In the basic usage, we can probably say that the library can be used in two ways:

* via Python API. Here we can then link the discoures post where we do more stuff)
* via CLI. Here we can list the 4 cli entrypoints that we provide, and say that by doing `python -m spark8t.cli.service_account_registry`, one can find more informations. We can then defer to the other Discourse posts, but saying that there these functionalities are exposed thought the SNAP

Reply:
Good point, adding it soon.

Reply:
Hmmm.
I'm a bit wondering whether it's beneficial to go into more technical details than what's there now (description of functionality + Discourse link).

The CLI commands are all accessed via aliases in our envs (Spark/ROCK). It sounds a little lame that we have 2-3 different ways to refer to them, depending on which docs you are reading :-)  (`spark-client.service-account-registry`, `service_account_registry.py`, etc.)
In case we'd unify the naming (typically using stuff like [console_scripts](https://packaging.python.org/en/latest/specifications/entry-points/) ), then this could make sense. But now I think it would just be a bit confusing.

On top of that, it would be just the duplication of Discourse. If we change the name of a script, we'd need to adjust documentiation on more-and-more places.

So on one hand I do see your point, but in my opinion with our current layout it wouldn't be elegant -- may even be contra-productive...